aws-xray-recorder-sdk-core/src/main/java/com/amazonaws/xray/entities/EntityImpl.java

```diff
@@ -586,39 +722,47 @@ public boolean isEmitted() {
      */
     @Override
     public void setEmitted(boolean emitted) {
-        checkAlreadyEmitted();
-        this.emitted = emitted;
+        synchronized (lock) {
+            checkAlreadyEmitted();
+            this.emitted = emitted;
+        }
     }
 
     @Override
     public String serialize() {
-        try {
-            return mapper.writeValueAsString(this);
-        } catch (JsonProcessingException jpe) {
-            logger.error("Exception while serializing entity.", jpe);
+        synchronized (lock) {
+            try {
+                return mapper.writeValueAsString(this);
+            } catch (JsonProcessingException jpe) {
+                logger.error("Exception while serializing entity.", jpe);
+            }
+            return "";
         }
-        return "";
     }
 
     @Override
     public String prettySerialize() {
-        try {
-            return mapper.writerWithDefaultPrettyPrinter().writeValueAsString(this);
-        } catch (JsonProcessingException jpe) {
-            logger.error("Exception while serializing segment.", jpe);
+        synchronized (lock) {
+            try {
+                return mapper.writerWithDefaultPrettyPrinter().writeValueAsString(this);
+            } catch (JsonProcessingException jpe) {
+                logger.error("Exception while serializing segment.", jpe);
+            }
+            return "";
         }
-        return "";
     }
 
     @Override
     public void removeSubsegment(Subsegment subsegment) {
-        getSubsegmentsLock().lock();
-        try {
-            subsegments.remove(subsegment);
-        } finally {
-            getSubsegmentsLock().unlock();
+        synchronized (lock) {
+            getSubsegmentsLock().lock();
```

Comment:
Do we still need to lock the subsegments list when we're locking on the entire parent entity?